algorithms/linfa-bayes/src/bernoulli_nb.rs

```diff
@@ -0,0 +1,336 @@
+use linfa::dataset::{AsSingleTargets, DatasetBase, Labels};
+use linfa::traits::{Fit, FitWith, PredictInplace};
+use linfa::{Float, Label};
+use ndarray::{Array1, ArrayBase, ArrayView2, Axis, Data, Ix2, OwnedRepr};
+use std::collections::HashMap;
+use std::hash::Hash;
+
+use crate::base_nb::{filter, NaiveBayes, NaiveBayesValidParams};
+use crate::error::{NaiveBayesError, Result};
+use crate::hyperparams::{BernoulliNbParams, BernoulliNbValidParams};
+use crate::multinomial_nb::MultinomialClassInfo;
+
+impl<'a, F, L, D, T> NaiveBayesValidParams<'a, F, L, D, T> for BernoulliNbValidParams<F, L>
+where
+    F: Float,
+    L: Label + 'a,
+    D: Data<Elem = F>,
+    T: AsSingleTargets<Elem = L> + Labels<Elem = L>,
+{
+}
+
+impl<F, L, D, T> Fit<ArrayBase<D, Ix2>, T, NaiveBayesError> for BernoulliNbValidParams<F, L>
+where
+    F: Float,
+    L: Label + Ord,
+    D: Data<Elem = F>,
+    T: AsSingleTargets<Elem = L> + Labels<Elem = L>,
+{
+    type Object = BernoulliNb<F, L>;
+    // Thin wrapper around the corresponding method of NaiveBayesValidParams
+    fn fit(&self, dataset: &DatasetBase<ArrayBase<D, Ix2>, T>) -> Result<Self::Object> {
+        let model = NaiveBayesValidParams::fit(self, dataset, None)?;
+        Ok(model.unwrap())
+    }
+}
+
+impl<'a, F, L, D, T> FitWith<'a, ArrayBase<D, Ix2>, T, NaiveBayesError>
+    for BernoulliNbValidParams<F, L>
+where
+    F: Float,
+    L: Label + 'a,
+    D: Data<Elem = F>,
+    T: AsSingleTargets<Elem = L> + Labels<Elem = L>,
+{
+    type ObjectIn = Option<BernoulliNb<F, L>>;
+    type ObjectOut = Option<BernoulliNb<F, L>>;
```

Comment:
This doesn't need to be an `Option`

Reply:
NVM I see that the rest of the `linfa-bayes` types also return `Option`s. We could fix this later but I'm not sure, so just leave this as is.